.github/copilot-instructions.md

```diff
@@ -0,0 +1,150 @@
+4. **Publish to local Maven**: `./gradlew publishToMavenLocal` (recommended for testing changes)
+
+#### Testing Commands:
+- **Run JVM tests for a module**: `./gradlew :kotlin-common-[module]:jvmTest`
+- **Run JS tests for a module**: `./gradlew :kotlin-common-[module]:jsTest` 
+- **Run all tests for a module**: `./gradlew :kotlin-common-[module]:allTests`
+- **Global test command**: There is no single `test` task - use `check` for comprehensive verification
+
+#### Documentation:
+- **Generate API docs**: `./gradlew :dokkaGeneratePublicationHtml`
+- **Output location**: `build/dokka/html/`
+
+### Build Timing and Known Issues
+
+**Timing Expectations**:
+- First build: 5-10 minutes (includes dependency resolution)
+- Subsequent builds: 30 seconds - 2 minutes
+- Test execution: 2-10 minutes depending on scope
+- Clean builds: Add 1-2 minutes
+
+**Common Warnings (can be ignored)**:
+- iOS target warnings on non-macOS systems
+- WebAssembly environment selection warnings
+- Dokka experimental plugin warnings
+- Gradle deprecation warnings (project targets Gradle 8.11.1)
+
+**Error Handling**:
+- If build fails with "Cannot locate tasks that match 'test'", use target-specific tasks like `jvmTest`, `jsTest`
+- Gradle daemon issues: Use `./gradlew --stop` then retry
+- Memory issues: Increase heap size in `gradle.properties` (currently set to 4GB for CI)
+
+## Project Layout and Architecture
+
+### Root Structure
+```
+/
+├── .github/workflows/        # CI/CD pipelines
+│   ├── kotlin-multiplatform-ci.yml    # Main CI (runs on every push)
+│   ├── dokka-gh-pages.yml             # Documentation deployment 
+│   └── copilot-setup-steps.yml        # Setup validation
+├── buildSrc/                # Custom Gradle build logic
+│   ├── build.gradle.kts     # Build dependencies and plugins
+│   └── src/                 # Convention plugins
+├── core/                    # Core Kotlin extensions (main module)
+├── [library-name]/          # Extension library modules
+├── build.gradle.kts         # Root build configuration
+├── settings.gradle.kts      # Project structure definition
+├── gradle.properties        # Build properties and JVM settings
+└── README.md               # Project documentation
+```
+
+### Key Modules
+- **core**: Core Kotlin language and stdlib extensions
+- **arrow**: Extensions for Arrow functional programming library
+- **coroutines**: Kotlin Coroutines utilities
+- **exposed**: Database library (Exposed) extensions  
+- **ktor**: HTTP client/server framework extensions
+- **net**: Network-related utilities
+- **reflect**: Kotlin reflection utilities
+- **serialization**: Kotlinx Serialization extensions
+- **vertx**: Vert.x framework extensions
+- **web**: Web-related common code
+
+### Configuration Files
+- **Build**: `build.gradle.kts` (root), `settings.gradle.kts`, `gradle.properties`
+- **CI/CD**: `.github/workflows/*.yml`
+- **Dependencies**: `buildSrc/build.gradle.kts` defines shared dependencies
```

Comment:
No this is incorrect or inaccurate. `buildSrc/build.gradle.kts` defines dependencies for `buildSrc` / the meta build / the Gradle classpath (you may find a more accurate term to describe this), `buildSrc/src/main/kotlin/VersionsAndDependencies.kt` defines shared dependencies for compilation and etc.